src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -796,17 +811,43 @@ def predict_real_time(self, test_data, accept="application/x-parquet"):
 
         Returns
         -------
-        Pandas.DataFrame
-        Predict results in DataFrame
+        Pandas.Series
+        Predict results in Series
         """
-        assert self.endpoint, "Please call `deploy()` to deploy an endpoint first."
-        assert accept in VALID_ACCEPT, f"Invalid accept type. Options are {VALID_ACCEPT}."
-        if type(test_data) == str:
-            test_data = load_pd.load(test_data)
-        if not isinstance(test_data, pd.DataFrame):
-            raise ValueError("test_data must be either a pandas.DataFrame, a local path or a s3 path")
+        self._validate_predict_real_time_args(accept)
+        test_data = self._load_predict_real_time_test_data(test_data)
+        pred, _ = self._predict_real_time(test_data=test_data, accept=accept)
+
+        return pred
+
+    def predict_proba_real_time(self, test_data, accept="application/x-parquet"):
```

Comment:
add type hints for test_data and outputs

Reply:
This is a nice suggestion. I think this is missing in general in APIs of the cloud module. Will have a PR dedicated to improve this instead